src/middlewares/roleMiddelware.ts

```diff
@@ -0,0 +1,37 @@
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+
+import { secret } from '../config/tokenKey.js';
+
+const roleMiddelware = (roles: string) => {
+    return function (req: Request, res: Response, next: NextFunction) {
+        if (req.method === 'OPTIONS') {
+            next();
+        }
+
+        try {
+            let token = req.headers.authorization;
+            if (token) {
+                token = token.split(' ')[1];
+            } else {
+                return res.status(403).json({ message: 'User is not authorized' });
+            }
+            const { roles: userRoles } = jwt.verify(token, secret) as JwtPayload;
```

Comment:
Separate roleMiddleware into two `authenticationMiddleware` and `roleMiddleware`
And after that use them in order `authenticationMiddleware` -> `roleMiddleware`

In `authenticationMiddleware` [extend](https://blog.logrocket.com/extend-express-request-object-typescript/) the request object with additional data from the parsed token